inst/shiny/modules/tab_nca/setup.R

```diff
@@ -0,0 +1,141 @@
+      ),
+      fluidRow(units_table_ui(ns("units_table"))),
+      settings_ui(ns("nca_settings"))
+    ),
+    nav_panel("Slope Selector", slope_selector_ui(ns("slope_selector"))),
+    nav_panel("Summary", summary_ui(ns("nca_setup_summary")))
+  )
+}
+
+setup_server <- function(id, data, adnca_data) {
+  moduleServer(id, function(input, output, session) {
+    # Gather all settings from the appropriate module
+    settings <- settings_server("nca_settings", data, adnca_data, settings_override)
+
+    # Create processed data object with applied settings.
+
+    processed_pknca_data <- reactive({
+      req(adnca_data(), settings())
+      log_trace("Updating PKNCA::data object.")
+
+      processed_pknca_data <- PKNCA_update_data_object(
+        adnca_data = adnca_data(),
+        auc_data = settings()$partial_aucs,
+        method = settings()$method,
+        selected_analytes = settings()$analyte,
+        selected_dosno = settings()$doseno,
+        selected_pcspec = settings()$pcspec,
+        params = settings()$parameter_selection,
+        should_impute_c0 = settings()$data_imputation$impute_c0
+      )
+
+      # Show bioavailability widget if it is possible to calculate
+      if (processed_pknca_data$dose$data$std_route %>% unique() %>% length() == 2) {
+        shinyjs::show(selector = ".bioavailability-picker")
+      } else {
+        shinyjs::hide(selector = ".bioavailability-picker")
+      }
+
+      if (nrow(processed_pknca_data$intervals) == 0) {
+        showNotification(
+          "All intervals were filtered. Please revise your settings",
+          type = "warning",
+          duration = 10
+        )
+      }
+
+      processed_pknca_data
+    })
+
+    # Parameter unit changes option: Opens a modal message with a units table to edit
+    units_table_server("units_table", processed_pknca_data)
+
+    # Create version for slope plots
+    # Only parameters required for the slope plots are set in intervals
+    # NCA dynamic changes/filters based on user selections
+    slopes_pknca_data <- reactive({
+      req(adnca_data(), settings())
+      log_trace("Updating PKNCA::data object for slopes.")
+
+      PKNCA_update_data_object(
+        adnca_data = adnca_data(),
+        auc_data = settings()$partial_aucs,
+        method = settings()$method,
+        selected_analytes = settings()$analyte,
+        selected_dosno = settings()$doseno,
+        selected_pcspec = settings()$pcspec,
+        params = c("lambda.z.n.points", "lambda.z.time.first",
+                   "r.squared", "adj.r.squared", "tmax"),
+        should_impute_c0 = settings()$data_imputation$impute_c0
+      )
+    })
+
+    slope_rules <- slope_selector_server(
+      "slope_selector",
+      slopes_pknca_data,
+      manual_slopes_override
+    )
+
+    summary_server("nca_setup_summary", processed_pknca_data)
+
+    # Handle downloading and uploading settings
+    output$settings_download <- downloadHandler(
+      filename = "aNCA_app_setup.Rds",
```

Comment:
suggestion: add something to improve the identification of the file

```suggestion
      filename = paste0("aNCA_settings_", Sys.Date(), ".rds"), 
```